packages/maskbook/src/plugins/Wallet/SNSAdaptor/RiskWarningDialog/RiskWarningBoundary.tsx

```diff
@@ -0,0 +1,40 @@
+import type { PropsWithChildren } from 'react'
+import { useEffect, useRef, useState } from 'react'
+import { Box } from '@material-ui/core'
+import { useWalletRiskWarningDialog } from '../../hooks/useWalletRiskWarningDialog'
+
+interface RiskWarningWrapperProps extends PropsWithChildren<{}> {
+    pluginId: string
+}
+
+export const RiskWarningBoundary = ({ children, pluginId }: RiskWarningWrapperProps) => {
+    const ref = useRef(null)
+    const [clicked, setClicked] = useState(false)
+
+    const { isConfirmed: isRiskWarningConfirmed, setDialog: setRiskWarningDialog } = useWalletRiskWarningDialog()
+
+    useEffect(() => {
+        if (!isRiskWarningConfirmed || !clicked) return
+
+        const currentBoundaryElement = ref.current as unknown as HTMLDivElement
+        const childNode = currentBoundaryElement.childNodes.length > 0 ? currentBoundaryElement.childNodes[0] : null
+        if (!childNode) return
+
+        const clickableElement = (childNode?.nodeType === 3 ? ref.current : childNode) as HTMLElement
+        clickableElement?.click()
+    }, [isRiskWarningConfirmed])
+
+    const handleRiskWarning = (event: React.MouseEvent<HTMLElement>) => {
+        if (!isRiskWarningConfirmed) {
+            event.stopPropagation()
+            setRiskWarningDialog({ open: true, pluginId })
+            setClicked(true)
+        }
+    }
+
+    return (
+        <Box ref={ref} onClickCapture={handleRiskWarning}>
```

Comment:
Pay attention to style mutations. A new wrapper node may break original styles. Check each place where you use this component.